test/org/zaproxy/zap/extension/pscanrulesAlpha/XPoweredByHeaderInfoLeakScannerUnitTest.java

```diff
@@ -0,0 +1,111 @@
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import net.htmlparser.jericho.Source;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.utils.I18N;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+
+/**
+ * @author Vahid Rafiei (@vahid_r)
+ * Date 2016-07-06
+ */
+
+@RunWith(MockitoJUnitRunner.class)
+public class XPoweredByHeaderInfoLeakScannerUnitTest {
```

Comment:
@psiinon requiring unit tests for promotion is a good plan.

@vahidR you can just copy the class, no issue there (it would have to be copied even if the pull request was already merged).
Once the branches are merged into one we can take a closer look at how to better handle the helper test classes.